src/main/scala/EnsimeExtrasPlugin.scala

```diff
@@ -221,4 +230,38 @@ object EnsimeExtrasPlugin extends AutoPlugin {
     extracted.append(newSettings, state)
   }
 
+  def formatOnlyTask: Def.Initialize[InputTask[Unit]] = InputTask(
+    (s: State) => Def.spaceDelimited()
+  ) { (argTask: TaskKey[Seq[String]]) =>
+      (argTask, sourceDirectories, scalariformPreferences, scalaVersion, streams).map {
+        (files, dirs, preferences, version, s) =>
+          if (files.isEmpty) throw new IllegalArgumentException("needs a file")
+          files.foreach(arg => {
+            val input: File = file(arg).getCanonicalFile
+            val sourceDirs = dirs.map(_.getCanonicalFile)
+
+            val here = sourceDirs.exists { dir => input.getPath.startsWith(dir.getPath) }
```

Comment:
it would be good to refactor this logic out into shared functionality with the `compileOnly` task